### PR TITLE
README: Update it

### PR DESCRIPTION
**Pull request by hypothesist**
Including the logo and overhauling the entire README page.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,130 +1,117 @@
-You might also want to visit `our website <http://coala-analyzer.org/>`_.
+.. image:: https://cloud.githubusercontent.com/assets/7521600/15992701/ef245fd4-30ef-11e6-992d-275c5ca7c3a0.jpg
```
Comment (Adrianzatreanu) [inline README.rst]: AMAZING!

File: README.rst [hunk 2]
```diff
@@ -1,130 +1,117 @@
...
+
+coala: Language Independent Code Analysis
+-------------
+coala is a **language agnostic code analysis toolkit**. With coala, users
+can create rules and standards to which the code should conform to. coala
+has an **user-friendly interface** that is completely customizable. It can
+be used in any environment and is completely modularized.
+
+coala has a set of official bears (plugins) for 37 languages including
```
Comment (Adrianzatreanu) [inline README.rst]: nonono, again, mentioning number of languages, this requires extra maintainance, you sure we want that?

also, comma after ``languages`` :D

File: README.rst [hunk 3]
```diff
@@ -1,130 +1,117 @@
...
+**C/C++**, **Python**, **JavaScript**, **CSS** and many more in addition
```
Comment (Adrianzatreanu) [inline README.rst]: maybe rephrase: ``including popular languages, such as: x, y, z, and many more, in addition to some generic language-independent algorithms.``

File: README.rst (hunk 2 above)
Comment (hypothesist) [inline README.rst]: Ah this was written a few weeks ago. Agreed, I'll link to bear-docs instead.

File: README.rst [hunk 4]
```diff
@@ -1,130 +1,117 @@
...
 
-|asciicast|
+========
+Features
+========
 
-.. |asciicast| image:: https://asciinema.org/a/42968.png
-   :target: https://asciinema.org/a/42968?autoplay=1
-   :width: 100%
+* Out-of-the-box support for **37 languages** with built-in check routines.
```
Comment (Adrianzatreanu) [inline README.rst]: number here too :(

Comment (hypothesist) [inline README.rst]: Excuse here too :P I'll fix it :+1:

File: README.rst [hunk 5]
```diff
@@ -1,130 +1,117 @@
...
 
-Newcomers Guide and Getting Involved
-------------------------------------
+================
+Getting Involved
+================
 
-If you are new and would like to contribute, read our `Getting Involved Site
-<http://coala.readthedocs.org/en/latest/Getting_Involved/README.html>`__!
+If you would like to contribute, please take a look at our `Getting Involved <http://coala.readthedocs.io/en/latest/Getting_Involved/README.html>`__ page or ask us at our gitter channel. We appreciate any help!
```
Comment (Adrianzatreanu) [inline README.rst]: in this section, also mention something like: ``We also prove a newcomer guide, which helps you get started by fixing an issue alone: guide link`` 
http://coala.readthedocs.io/en/latest/Getting_Involved/Newcomers.html

File: README.rst [hunk 6]
```diff
@@ -1,130 +1,117 @@
...
-*coala* has a set of official bears (plugins) to provide an out-of-the-box
-analysis functionality for many popular languages in addition to some
-generically applicable algorithms.
+============
+Installation
+============
 
-To see what coala can do for you and your language, take a look at
-`our capabilities listing <https://github.com/coala-analyzer/bear-docs/blob/master/README.rst>`__.
+To install the **latest stable version** run:
```
Comment (Adrianzatreanu) [inline README.rst]: why is ``latest stable version`` bolded but ``latest development version`` isn't :D ?

File: README.rst [hunk 7]
```diff
@@ -1,130 +1,117 @@
...
-To install *coala* with the official set of analysis routines you can simply run
-``pip3 install coala-bears``.
+-----
 
-*coala* only, without the official bears, can be installed with
-``pip3 install coala``. If you need more information about the installation and
-dependencies, take a look at our `installation documentation
-<http://coala.rtfd.org/en/latest/Users/Install.html>`__.
+=====
+Usage
```
Comment (Adrianzatreanu) [inline README.rst]: a vague section. maybe give one or two examples? Such as: To run coala over your project, create a .coafile section (click here to see more about sections), or use our default one from the repo.

To run coala over a file with a specific bear, run coala --files x.py --bears fdkalbear

For a full list of bears, check out this link: bearslink.

File: README.rst (hunk 6 above)
Comment (hypothesist) [inline README.rst]: Because the first thing I look for when I land on a github page is how to install and I want the stable version most times. So this serves both purposes IMO.

File: README.rst [hunk 8]
```diff
@@ -1,130 +1,154 @@
...
+`click here. <https://github.com/coala-analyzer/bear-docs/blob/master/README.rst>`__
+
+|Linux Build Status| |Windows Build status| |Scrutinizer Code Quality|
+|codecov.io| |Documentation Status| |Gitmate|
 
 .. Start ignoring LineLengthBear
 
-::
+================================================= ================================================ =========================================================
+`Official Website <http://coala-analyzer.org/>`__ `Twitter <https://twitter.com/coala_analyzer>`__ `Video Demo <https://asciinema.org/a/42968?autoplay=1>`__
```
Comment (Adrianzatreanu) [inline README.rst]: https://www.facebook.com/coalaAnalyzer/?fref=ts we have this too :D

File: README.rst [hunk 9]
```diff
@@ -1,130 +1,154 @@
...
 
-Newcomers Guide and Getting Involved
-------------------------------------
+Please read our
+`coafile specification <http://coala.readthedocs.io/en/latest/Users/coafile.html>`__
+to learn more.
 
-If you are new and would like to contribute, read our `Getting Involved Site
-<http://coala.readthedocs.org/en/latest/Getting_Involved/README.html>`__!
+If you want to quickly want to test some settings without a ``.coafile``, you
```
Comment (Adrianzatreanu) [inline README.rst]: you've used ``want`` twice :D 

Hm i was thinking maybe something like this instead:

``If you want to quickly test some settings, without creating a ".coafile", you can also run coala in the command line and pass it arguments, such as:``

File: README.rst [hunk 10]
```diff
@@ -1,130 +1,154 @@
...
+
+================
+Getting Involved
+================
+
+If you would like to contribute, please take a look at our
+`Getting Involved <http://coala.readthedocs.io/en/latest/Getting_Involved/README.html>`__
+page or ask us at our gitter channel. We also have a
+`newcomer guide <http://coala.readthedocs.io/en/latest/Getting_Involved/Newcomers.html>`__
+to help you get started in fixing an issue. We appreciate any help!
```
Comment (Adrianzatreanu) [inline README.rst]: should be more accent on the achieving of the newcomer guide, as it's helping users fix an issue themselves :D

so something like: ``We also have a newcomer guide, which helps you get started by fixing an issue yourself! We appreciate any help!``

Comment (Adrianzatreanu) [inline README.rst]: actually i dont really like the whole section. let me rethink it a little

Comment (Adrianzatreanu) [inline README.rst]: If you would like to be part of the coalaian community, you can check our Getting Involved page, or simply join us on our gitter channel, where we are really active, with community members from all over the world, happy to answer any question at any time of day. (maybe drop the ``at any time of day`` part if you dont like it, its experimental :p)

We also have a newcomer guide, which helps you get started by fixing an issue yourself! We appreciate any help!
If you stumble upon anything, don't worry, just drop by and contact us by dropping a mail at our mailing list(link) or gitter(link), and you'll be in trouble no more!

Comment (Adrianzatreanu) [inline README.rst]: try this, more or less, i think it sounds way better than what we have :)

File: README.rst [hunk 11]
```diff
@@ -1,130 +1,154 @@
...
 
-|AGPL|
+=======
+Support
+=======
 
-This code falls under the GNU Affero General Public License as published
-by the Free Software Foundation, either version 3 of the License, or (at
-your option) any later version.
+Feel free to contact us at our `Gitter page <https://gitter.im/coala-analyzer/coala>`__, we'd be happy to help!
```
Comment (Adrianzatreanu) [inline README.rst]: mention mailing list, https://github.com/coala-analyzer/coala/wiki/Mailing-Lists

File: README.rst (hunk 9 above)
Comment (hypothesist) [inline README.rst]: I think it's implied they have to run in a shell, no?

File: README.rst [hunk 12]
```diff
@@ -1,130 +1,165 @@
-You might also want to visit `our website <http://coala-analyzer.org/>`_.
+.. image:: https://cloud.githubusercontent.com/assets/7521600/15992701/ef245fd4-30ef-11e6-992d-275c5ca7c3a0.jpg
+
+coala: Language Independent Code Analysis
+-------------
+**coala provides a common command-line interface for linting and fixing all
```
Comment (hypothesist) [inline README.rst]: @sils1297 This is from the engagement thing. IMO `provides an unified command-line...` sounds better.

File: README.rst [hunk 13]
```diff
@@ -1,130 +1,165 @@
...
+your code, regardless of the programming languages you use.**
+
+With coala, users can create rules and standards to be followed in the source
+code. coala has an **user-friendly interface** that is completely customizable.
+It can be used in any environment and is completely modularized.
```
Comment (Adrianzatreanu) [inline README.rst]: im not very sure if the correct word here is ``modular`` or ``modularized`` :-??

File: README.rst [hunk 14]
```diff
@@ -1,130 +1,165 @@
...
 
-|asciicast|
+========
+Features
+========
 
-.. |asciicast| image:: https://asciinema.org/a/42968.png
-   :target: https://asciinema.org/a/42968?autoplay=1
-   :width: 100%
+* Out-of-the-box support for various `popular languages <https://github.com/coala-analyzer/bear-docs/blob/master/README.rst>`__
```
Comment (Adrianzatreanu) [inline README.rst]: comma before ``such as``

File: README.rst [hunk 15]
```diff
@@ -1,130 +1,165 @@
-You might also want to visit `our website <http://coala-analyzer.org/>`_.
+.. image:: https://cloud.githubusercontent.com/assets/7521600/15992701/ef245fd4-30ef-11e6-992d-275c5ca7c3a0.jpg
+
+coala: Language Independent Code Analysis
+-------------
+**coala provides a common command-line interface for linting and fixing all
+your code, regardless of the programming languages you use.**
+
+With coala, users can create rules and standards to be followed in the source
```
Comment (Adrianzatreanu) [inline README.rst]: maybe we should link them to how to create a coafile so they can reed more there if interested how they create rules and standards? have it: ``can create `rules and standards <link>`__ to be..``

Comment (Adrianzatreanu) [inline README.rst]: http://coala.readthedocs.io/en/latest/Users/coafile.html is the link btw

File: README.rst [hunk 16]
```diff
@@ -1,130 +1,165 @@
...
+    :depth: 1
+    :backlinks: none
 
-Demo (Click to View)
---------------------
+-----
 
-|asciicast|
+========
+Features
```
Comment (Adrianzatreanu) [inline README.rst]: add that its quite fast? maybe some timers even? with caching now? imo a few seconds on a huge project like coala is something not many projects achieve

File: README.rst [hunk 17]
```diff
@@ -1,130 +1,165 @@
...
+========
 
-.. |asciicast| image:: https://asciinema.org/a/42968.png
-   :target: https://asciinema.org/a/42968?autoplay=1
-   :width: 100%
+* Out-of-the-box support for various `popular languages <https://github.com/coala-analyzer/bear-docs/blob/master/README.rst>`__
+  such as **C/C++**, **Python**, **Javascript**, **CSS**, **Java** and many
+  others with built-in check routines.
+* User-friendly interfaces such as JSON, interactive CLI or any custom format.
+* Plugins for gedit, Sublime Text, Atom, Vim and Emacs.
```
Comment (Adrianzatreanu) [inline README.rst]: im a huge fan of documentations that have many helpful clickable links, not sure others agree, but id love if you could give an official link for each editor here, so people would click on it and know what that is :D ?

or even to the coala-gedit/coala-atom repo? i dont know, whatever you feel its better :)

File: README.rst [hunk 18]
```diff
@@ -1,130 +1,165 @@
...
 
-*coala* is a language independent analysis toolkit. It empowers developers
-to create rules which a project's code should conform to. *coala* takes care
-of showing these issues to users in a friendly manner, is versatile and can be
-used in any environment. Patches to automatically fix code will be managed too.
-*coala* has a set of official bears (plugins) to provide an out-of-the-box
-analysis functionality for many popular languages in addition to some
-generically applicable algorithms.
+============
+Installation
```
Comment (Adrianzatreanu) [inline README.rst]: not sure if this is the section to do that, but maybe we should have the latest stable version here noted somewhere? id love to see it here, and its not hard maintaining, we have a new version each month or so.

also, not sure this is doable since this is rst, but do you think theres a way to generate here the latest development version? like the one we have when we import VERSION? :D

File: README.rst [hunk 19]
```diff
@@ -1,130 +1,165 @@
...
 
-To see what coala can do for you and your language, take a look at
-`our capabilities listing <https://github.com/coala-analyzer/bear-docs/blob/master/README.rst>`__.
+To install the **latest stable version** run:
 
-*coala* is written with a lower case "c".
+.. code-block:: bash
 
-Read more at our `documentation <http://coala.rtfd.org/>`__.
+    $ pip3 install coala-bears
```
Comment (Adrianzatreanu) [inline README.rst]: this might be confusing for users. why the hell would they install ``coala-bears`` when they are on ``coala`` page?
should mention "To simply install coala and its basic functionalities, pip install coala. To also install the bears and bla bla bla, install coala-bears``. or even if you dont mention the first installation command, with plain coala, you should mention why the hell they should install coala-bears and not coala :P

File: README.rst [hunk 20]
```diff
@@ -1,130 +1,165 @@
...
 
-Project Status
---------------
+.. code-block:: bash
 
-|Linux Build Status| |Windows Build status|
+    [Default]
+    files = src/**/*.py
+    bears = SpaceConsistencyBear
+    use_spaces = True
```
Comment (Adrianzatreanu) [inline README.rst]: this section could use some love. I mean, here's how we teach newcomers how to use coala, eh?

first of all, dont name it Default. secondly, explain it after, briefly.

[Spacing]
files = src/**/*.py
bears = SpaceConsistencyBear
use_spaces = True

and then explain:
``Where Spacing is the name of the section, files is the files you would like to run coala on, bears are the bears that are to be used, and use_spaces specifies the bear to use spaces instead of tabs.
After writing this section, simply run "$ coala spacing" and you are done.``

Comment (Adrianzatreanu) [inline README.rst]: this readme section should basically teach someone, who has NO IDEA what coala is or how it is used, to be able to run coala on whatever file he wants, with a bear and a setting :D he should understand what each does, and why he does it like that

File: README.rst [hunk 21]
```diff
@@ -1,130 +1,165 @@
...
-<http://coala.readthedocs.org/en/latest/Getting_Involved/README.html>`__!
+Please read our
+`coafile specification <http://coala.readthedocs.io/en/latest/Users/coafile.html>`__
+to learn more.
 
-We appreciate any help! Feel free to message us on
-`gitter <https://gitter.im/coala-analyzer/coala>`__. If you have any
-questions we're happy to help you!
+If you want to quickly test some settings without a ``.coafile``, you
+can run:
```
Comment (Adrianzatreanu) [inline README.rst]: try something like: ``However, if you wouldn't like to save your settings, you can also run coala with some arguemnts from the command line, such as:``

 $ coala --files=setup.py --bears=SpaceConsistencyBear -S use_spaces=True

``Note that this code does the same thing as the one above.``

File: README.rst [hunk 22]
```diff
@@ -1,130 +1,165 @@
...
 
-Please note that some files or content may be copied from other places.
-Most of them are GPL compatible. There is a small portion of code in the
-tests that falls under the Creative Commons license, see
-https://creativecommons.org/licenses/by-sa/3.0/deed.de for more
-information.
+-----
+
+================
+Getting Involved
```
Comment (Adrianzatreanu) [inline README.rst]: imo this section should be before ``Support``, that one is kind of what youd want to see at the end of the file. dont forget to swap in the toctree too, up there.

File: README.rst (hunk 17 above)
Comment (hypothesist) [inline README.rst]: I think a link to the coala-gedit repo is better since a gedit user has no use to a link to the gedit home page :P

File: README.rst [hunk 23]
```diff
@@ -1,130 +1,165 @@
...
+If you would like to be a part of the coala community, you can check out our
+`Getting Involved <http://coala.readthedocs.io/en/latest/Getting_Involved/README.html>`__
+page or ask us at our active Gitter channel, where we have maintainers from
+all over the world. We appreciate any help!
+
+We also have a
+`newcomer guide <http://coala.readthedocs.io/en/latest/Getting_Involved/Newcomers.html>`__
+to help you get started by fixing an issue yourself! If you get stuck anywhere
+or need some help, feel free to drop a mail at our
+`newcomer mailing list <https://groups.google.com/d/forum/coala-newcomers>`__
```
Comment (Adrianzatreanu) [inline README.rst]: maybe the order should be reversed? "contact us on gitter or drop a mail"? i think our strong point is gitter

File: README.rst (hunk 16 above)
Comment (hypothesist) [inline README.rst]: There are already two points below for performance - one about multi-threading and the other about caching. What do you have in mind?

File: README.rst [hunk 24]
```diff
@@ -1,130 +1,165 @@
...
+
+|gitter|
+
+-----
+
+=======
+Authors
+=======
+
+coala is maintained by a growing community. Please take a look at the
```
Comment (Adrianzatreanu) [inline README.rst]: this is super unhelpful. also that meta information is quite old, its only 3 guys in there and you say ``a growing community``. this is misleading :D again im asking as above, is there a method to list them here automatically, all the guys from coala, ordered by commits or something, like we have on the website, or it is too much :D ? or maybe link them to the page where you see the commits order? i just think that this information is vague :D

File: README.rst [hunk 25]
```diff
@@ -137,9 +172,10 @@ information.
    :target: https://codecov.io/github/coala-analyzer/coala?branch=master
 .. |Documentation Status| image:: https://readthedocs.org/projects/coala/badge/?version=latest
    :target: http://coala.rtfd.org/
-.. |https://gitter.im/coala-analyzer/coala| image:: https://img.shields.io/badge/gitter-join%20chat%20%E2%86%92-brightgreen.svg
-   :target: https://gitter.im/coala-analyzer/coala
 .. |AGPL| image:: https://img.shields.io/github/license/coala-analyzer/coala.svg
    :target: https://www.gnu.org/licenses/agpl-3.0.html
 .. |Gitmate| image:: https://img.shields.io/badge/Gitmate-0%20issues-brightgreen.svg
    :target: http://gitmate.com/
+.. |gitter| image:: https://badges.gitter.im/coala-analyzer/coala.svg
```
Comment (Adrianzatreanu) [inline README.rst]: why does this gitter thingie appear on both Support and Getting Involved sections? looks kinda ugly. should keep only one of them maybe :D ?

File: README.rst [hunk 26]
```diff
@@ -1,130 +1,165 @@
...
 
-*coala* only, without the official bears, can be installed with
-``pip3 install coala``. If you need more information about the installation and
-dependencies, take a look at our `installation documentation
-<http://coala.rtfd.org/en/latest/Users/Install.html>`__.
+=====
+Usage
+=====
 
-The latest code from master is automatically deployed to PyPI as a
```
Comment (Adrianzatreanu) [inline README.rst]: this is a cool part taht you could have kept in your new  design? its good to know this :D

File: README.rst [hunk 27]
```diff
@@ -1,130 +1,165 @@
...
 
-- *coala* provides **built-in checking routines** (named bears in *coala*).
-- **Serves your requirement**: You can easily write your own checks (using
-  bears).
-- *coala* provides **user-friendly interfaces** like json, formatted and
-  interactive output in the CLI and *plugins for various editors* are
-  available as well.
-- **Optimal performance**: *coala* manages parallelizing the checking-routines
-  without you having to worry.
-- **Unified interface**: One tool for all programming languages.
```
Comment (Adrianzatreanu) [inline README.rst]: one tool to rule them all :D maybe mention this?

File: README.rst (hunk 18 above)
Comment (hypothesist) [inline README.rst]: We can easily update it manually - as you said it's just a monthly task.

File: README.rst [hunk 28]
```diff
@@ -1,130 +1,165 @@
...
-of showing these issues to users in a friendly manner, is versatile and can be
-used in any environment. Patches to automatically fix code will be managed too.
-*coala* has a set of official bears (plugins) to provide an out-of-the-box
-analysis functionality for many popular languages in addition to some
-generically applicable algorithms.
+============
+Installation
+============
 
-To see what coala can do for you and your language, take a look at
```
Comment (Adrianzatreanu) [inline README.rst]: i think we have a command that lists all bears for a particular language. and should be listed, such as: To see what coala can do for your language, type coala "command" python".

File: README.rst (hunk 16 above)
Comment (Adrianzatreanu) [inline README.rst]: give an example?

usually people are impressed by concrete examples. 

"coala can fix a 50 files repo in 5 seconds" or something like this :D

my point is: if you tell someone you are fast, they wont be impressed. if you tell someone you can run 50m in 8 seconds, they will be impressed, even if this is a quite slow time :) thats my point

File: README.rst (hunk 17 above)
Comment (Adrianzatreanu) [inline README.rst]: great!

File: README.rst (hunk 13 above)
Comment (hypothesist) [inline README.rst]: I think it's modular :+1:

File: README.rst (hunk 22 above)
Comment (hypothesist) [inline README.rst]: the contents are generated automatically :P

File: README.rst (hunk 25 above)
Comment (hypothesist) [inline README.rst]: which one?

File: README.rst (hunk 27 above)
Comment (hypothesist) [inline README.rst]: Put that in the image?

File: README.rst [hunk 29]
```diff
@@ -1,132 +1,196 @@
...
-- **Unified interface**: One tool for all programming languages.
+    [Spacing]
+    files = src/**/*.py
+    bears = SpaceConsistencyBear
+    use_spaces = True
 
-If you want to learn more about *coala*, its functionality and its usage,
-please take a look at our
-`tutorial <http://coala.rtfd.org/en/latest/Users/Tutorials/Tutorial.html>`__.
+* The `files` key tells coala which files to lint - here, we're linting all
```
Comment (Adrianzatreanu) [inline README.rst]: double backticks

Comment (Adrianzatreanu) [inline README.rst]: comma isnt needed here.

File: README.rst [hunk 30]
```diff
@@ -1,132 +1,196 @@
-You might also want to visit `our website <http://coala-analyzer.org/>`_.
+.. image:: https://cloud.githubusercontent.com/assets/7521600/15992701/ef245fd4-30ef-11e6-992d-275c5ca7c3a0.jpg
+
+coala: Language Independent Code Analysis
+-------------
```
Comment (sils1297) [inline README.rst]: add newline here, also please continue the dashes for the length of the heading

File: README.rst (hunk 12 above)
Comment (sils1297) [inline README.rst]: then use unified

File: README.rst [hunk 31]
```diff
@@ -1,132 +1,196 @@
...
+    [Spacing]
+    files = src/**/*.py
+    bears = SpaceConsistencyBear
+    use_spaces = True
 
-If you want to learn more about *coala*, its functionality and its usage,
-please take a look at our
-`tutorial <http://coala.rtfd.org/en/latest/Users/Tutorials/Tutorial.html>`__.
+* The `files` key tells coala which files to lint - here, we're linting all
+  python files inside the `src` directory.
```
Comment (Adrianzatreanu) [inline README.rst]: here we are using a glob to analyze all the python files under the ``src/`` directory.

File: README.rst [hunk 32]
```diff
@@ -1,132 +1,196 @@
-You might also want to visit `our website <http://coala-analyzer.org/>`_.
+.. image:: https://cloud.githubusercontent.com/assets/7521600/15992701/ef245fd4-30ef-11e6-992d-275c5ca7c3a0.jpg
+
+coala: Language Independent Code Analysis
+-------------
+**coala provides a common command-line interface for linting and fixing all
+your code, regardless of the programming languages you use.**
+
+With coala, users can create
+`rules and standards <http://coala.readthedocs.io/en/latest/Users/coafile.html>`__
```
Comment (sils1297) [inline README.rst]: not sure we should point users there, CC @Adrianzatreanu maybe even remove that doc page? Or make it hold better info with useful examples.

File: README.rst [hunk 33]
```diff
@@ -1,132 +1,196 @@
...
+    files = src/**/*.py
+    bears = SpaceConsistencyBear
+    use_spaces = True
 
-If you want to learn more about *coala*, its functionality and its usage,
-please take a look at our
-`tutorial <http://coala.rtfd.org/en/latest/Users/Tutorials/Tutorial.html>`__.
+* The `files` key tells coala which files to lint - here, we're linting all
+  python files inside the `src` directory.
+* The `bears` key specifies which bears (plugins) to use. We support a huge
```
Comment (Adrianzatreanu) [inline README.rst]: which bears you want to use. **

File: README.rst [hunk 34]
```diff
@@ -1,132 +1,196 @@
...
+  number of languages and you can find the whole list
+  `here <github.com/coala-analyzer/bear-docs>`__.
```
Comment (Adrianzatreanu) [inline README.rst]: If your language is not there, don't worry, we got some bears that work for any language (link "bears that work for any language" to the bear docs section with the general bears) or you can file an issue (https://github.com/coala-analyzer/coala-bears/issues/new) and we will do that for you!

File: README.rst [hunk 35]
```diff
@@ -1,132 +1,196 @@
...
 
-*coala* only, without the official bears, can be installed with
-``pip3 install coala``. If you need more information about the installation and
-dependencies, take a look at our `installation documentation
-<http://coala.rtfd.org/en/latest/Users/Install.html>`__.
+    $ coala
 
-The latest code from master is automatically deployed to PyPI as a
-development version. Get it with ``pip3 install coala --pre``.
+Please read our
```
Comment (Adrianzatreanu) [inline README.rst]: First mention that:

``It will run all the defined sections in the order you defined them.``

File: README.rst [hunk 36]
```diff
@@ -1,132 +1,196 @@
...
+`coafile specification <http://coala.readthedocs.io/en/latest/Users/coafile.html>`__
+to learn more.
 
-|PyPI|
+However, if you don't want to save your settings, you can also run coala with
+command lind arguments:
```
Comment (Adrianzatreanu) [inline README.rst]: line**

File: README.rst [hunk 37]
```diff
@@ -1,132 +1,196 @@
...
-Authors
--------
+.. code-block:: bash
 
-*coala* is maintained by a growing community. Please take a look at the
-meta information in `setup.py <setup.py>`__ for current maintainers.
+    $ coala --files=setup.py --bears=SpaceConsistencyBear -S use_spaces=True
+
+Note that this command does the same thing as having a coafile and running
+`coala`.
```
Comment (Adrianzatreanu) [inline README.rst]: ``The only difference is that having a coafile will save the coafile for other uses.``

File: README.rst [hunk 38]
```diff
@@ -1,132 +1,196 @@
...
+`Getting Involved <http://coala.readthedocs.io/en/latest/Getting_Involved/README.html>`__
+page or ask us at our active Gitter channel, where we have maintainers from
+all over the world. We appreciate any help!
 
-|Documentation Status| |Gitmate|
+We also have a
+`newcomer guide <http://coala.readthedocs.io/en/latest/Getting_Involved/Newcomers.html>`__
+to help you get started by fixing an issue yourself! If you get stuck anywhere
+or need some help, feel free to contact us on Gitter or drop a mail at our
+`newcomer mailing list <https://groups.google.com/d/forum/coala-newcomers>`__
```
Comment (Adrianzatreanu) [inline README.rst]: end with full stop

File: README.rst [hunk 39]
```diff
@@ -1,132 +1,196 @@
...
-Most of them are GPL compatible. There is a small portion of code in the
-tests that falls under the Creative Commons license, see
-https://creativecommons.org/licenses/by-sa/3.0/deed.de for more
-information.
+=======
+Authors
+=======
+
+coala is maintained by a growing community. Please take a look at the
+meta information in `setup.py <setup.py>`__ for current maintainers.
```
Comment (Adrianzatreanu) [inline README.rst]: for the *

File: README.rst (hunk 32 above)
Comment (hypothesist) [inline README.rst]: Well we definitely need a proper coafile documentation

File: README.rst [hunk 40]
```diff
@@ -1,132 +1,200 @@
...
+
+With coala, users can create
+`rules and standards <http://coala.readthedocs.io/en/latest/Users/coafile.html>`__
+to be followed in the source
+code. coala has an **user-friendly interface** that is completely customizable.
+It can be used in any environment and is completely modular.
+
+coala has a set of official bears (plugins) for several languages, including
+popular languages such as **C/C++**, **Python**, **JavaScript**, **CSS**,
+**Java** and many more, in addition to some generic language independent
```
Comment (Adrianzatreanu) [inline README.rst]: language-independent. you need a dash to show what is independent :p

File: README.rst [hunk 41]
```diff
@@ -1,132 +1,200 @@
...
+algorithms. To learn more about the different languages supported and the
```
Comment (Adrianzatreanu) [inline README.rst]: maybe move the last sentence on a newline with a newline between the paragraphs to emphasize it

File: README.rst [hunk 42]
```diff
@@ -1,132 +1,200 @@
...
+-----
+
+========
+Features
+========
+
+* Out-of-the-box support for various `popular languages <https://github.com/coala-analyzer/bear-docs/blob/master/README.rst>`__,
+  such as **C/C++**, **Python**, **Javascript**, **CSS**, **Java** and many
+  others with built-in check routines.
+* User-friendly interfaces such as JSON, interactive CLI or any custom format.
```
Comment (Adrianzatreanu) [inline README.rst]: i think a comma is needed after interfaces

File: README.rst [hunk 43]
```diff
@@ -1,132 +1,200 @@
...
-config!
+.. code-block:: bash
 
-Demo (Click to View)
---------------------
+    $ pip3 install coala-bears --pre
 
-|asciicast|
+The latest code from the master branch is automatically deployed as the
+development version in PyPI.
```
Comment (Adrianzatreanu) [inline README.rst]: i know i suggested this, but i think it should be ``a development version``.. :( sorry

File: README.rst [hunk 44]
```diff
@@ -1,132 +1,200 @@
...
-If you want to learn more about *coala*, its functionality and its usage,
-please take a look at our
-`tutorial <http://coala.rtfd.org/en/latest/Users/Tutorials/Tutorial.html>`__.
+* The ``files`` key tells coala which files to lint - here we're linting all
+  python files inside the `src/` directory by using a glob expression.
+* The ``bears`` key specifies which bears (plugins) you want to use. We support
+  a huge number of languages and you can find the whole list
+  `here <github.com/coala-analyzer/bear-docs/blob/master/README.rst>`__.
+  If you don't find your langauge there, we've got some
+  `bears that work for all languages <https://github.com/coala-analyzer/bear-docs/blob/master/README.rst#all>`__. Or you can file an issue and we would build a bear for you!
```
Comment (Adrianzatreanu) [inline README.rst]: not ``build``, but ``create``.

Comment (Adrianzatreanu) [inline README.rst]: replace Or with Additionally

File: README.rst [hunk 45]
```diff
@@ -1,132 +1,200 @@
...
+    [Spacing]
+    files = src/**/*.py
+    bears = SpaceConsistencyBear
+    use_spaces = True
 
-If you want to learn more about *coala*, its functionality and its usage,
-please take a look at our
-`tutorial <http://coala.rtfd.org/en/latest/Users/Tutorials/Tutorial.html>`__.
+* The ``files`` key tells coala which files to lint - here we're linting all
+  python files inside the `src/` directory by using a glob expression.
```
Comment (Adrianzatreanu) [inline README.rst]: double backticks not just one on src/

File: README.rst [hunk 46]
```diff
@@ -1,132 +1,200 @@
...
 
-*coala* is a language independent analysis toolkit. It empowers developers
-to create rules which a project's code should conform to. *coala* takes care
-of showing these issues to users in a friendly manner, is versatile and can be
-used in any environment. Patches to automatically fix code will be managed too.
-*coala* has a set of official bears (plugins) to provide an out-of-the-box
-analysis functionality for many popular languages in addition to some
-generically applicable algorithms.
+=====
+Usage
```
Comment (Adrianzatreanu) [inline README.rst]: this section requires the most care imo, im sorry for making you go over this again but i think you could modify it a little.
Try making it a step-based guide.

Comment (Adrianzatreanu) [inline README.rst]: first of all move the coala --help thingie to the end, nobody uses that, they want to follow the guide

Comment (Adrianzatreanu) [inline README.rst]: "To run coala, you have two options: you store a coafile and save some customizable sections, or simply run coala from the command line over the file(s) you give as an argument.``

Then you go here with a step based guide from what you already did (just number them: 1, 2, 3, etc, your guide is already cool, dont change much)

and then go with the other argument thingie :D for more info just ping me :)

**Comment by Adrianzatreanu** [comment]
also rebase needed. this is amazing, cant wait to get it merged :)

**Comment by Adrianzatreanu** [comment]
ack 77054df

**Comment by hypothesist** [comment]
reack https://github.com/coala-analyzer/coala/pull/2413/commits/868c926a4e24259e7f1224a6a0671d6b82f25b39

**Comment by hypothesist** [comment]
@rultor merge
